## Задание 1
### Считать данные с помощью pandas
### Вывести на экран первые 5 строк
### Посмотреть на описание признаков и на их содержание


In [1]:
import pandas as pd
df = pd.read_csv('kc-house-data.csv')
df.head(5)

id             date     price  bedrooms  bathrooms  sqft_living   
0  7129300520  20141013T000000  221900.0         3       1.00         1180  \
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement   
0      5650     1.0           0     0  ...      7        1180              0  \
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15   
0      1955             0    98178  47.5112 -122.257           1340  \
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [2]:
df.date = pd.to_datetime(df.date)

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   id             21613 non-null  int64         
 1   date           21613 non-null  datetime64[ns]
 2   price          21613 non-null  float64       
 3   bedrooms       21613 non-null  int64         
 4   bathrooms      21613 non-null  float64       
 5   sqft_living    21613 non-null  int64         
 6   sqft_lot       21613 non-null  int64         
 7   floors         21613 non-null  float64       
 8   waterfront     21613 non-null  int64         
 9   view           21613 non-null  int64         
 10  condition      21613 non-null  int64         
 11  grade          21613 non-null  int64         
 12  sqft_above     21613 non-null  int64         
 13  sqft_basement  21613 non-null  int64         
 14  yr_built       21613 non-null  int64         
 15  yr_renovated   2161

In [4]:
## Задание 2 _(первичный анализ данных)_
### 2.1 Изучите типы данных
### 2.2 Найдите количество пропущенных ячеек в данных
### 2.3 Посчитайте основные статистики по всем признакам и поизучайте их
### 2.4 Пишите выводы

In [5]:
df.dtypes

id                        int64
date             datetime64[ns]
price                   float64
bedrooms                  int64
bathrooms               float64
sqft_living               int64
sqft_lot                  int64
floors                  float64
waterfront                int64
view                      int64
condition                 int64
grade                     int64
sqft_above                int64
sqft_basement             int64
yr_built                  int64
yr_renovated              int64
zipcode                   int64
lat                     float64
long                    float64
sqft_living15             int64
sqft_lot15                int64
dtype: object

In [6]:
df.isnull().sum()

id               0
date             0
price            0
bedrooms         0
bathrooms        0
sqft_living      0
sqft_lot         0
floors           0
waterfront       0
view             0
condition        0
grade            0
sqft_above       0
sqft_basement    0
yr_built         0
yr_renovated     0
zipcode          0
lat              0
long             0
sqft_living15    0
sqft_lot15       0
dtype: int64

In [7]:
df.describe()

id                           date         price   
count  2.161300e+04                          21613  2.161300e+04  \
mean   4.580302e+09  2014-10-29 04:38:01.959931648  5.400881e+05   
min    1.000102e+06            2014-05-02 00:00:00  7.500000e+04   
25%    2.123049e+09            2014-07-22 00:00:00  3.219500e+05   
50%    3.904930e+09            2014-10-16 00:00:00  4.500000e+05   
75%    7.308900e+09            2015-02-17 00:00:00  6.450000e+05   
max    9.900000e+09            2015-05-27 00:00:00  7.700000e+06   
std    2.876566e+09                            NaN  3.671272e+05   

           bedrooms     bathrooms   sqft_living      sqft_lot        floors   
count  21613.000000  21613.000000  21613.000000  2.161300e+04  21613.000000  \
mean       3.370842      2.114757   2079.899736  1.510697e+04      1.494309   
min        0.000000      0.000000    290.000000  5.200000e+02      1.000000   
25%        3.000000      1.750000   1427.000000  5.040000e+03      1.000000   
50%        3.000000      2.250000   1910.000000  7.618000e+03      1.500000   
75%        4.000000      2.500000   2550.000000  1.068800e+04      2.000000   
max       33.000000      8.000000  13540.000000  1.651359e+06      3.500000   
std        0.930062      0.770163    918.440897  4.142051e+04      0.539989   

         waterfront          view  ...         grade    sqft_above   
count  21613.000000  21613.000000  ...  21613.000000  21613.000000  \
mean       0.007542      0.234303  ...      7.656873   1788.390691   
min        0.000000      0.000000  ...      1.000000    290.000000   
25%        0.000000      0.000000  ...      7.000000   1190.000000   
50%        0.000000      0.000000  ...      7.000000   1560.000000   
75%        0.000000      0.000000  ...      8.000000   2210.000000   
max        1.000000      4.000000  ...     13.000000   9410.000000   
std        0.086517      0.766318  ...      1.175459    828.090978   

       sqft_basement      yr_built  yr_renovated       zipcode           lat   
count   21613.000000  21613.000000  21613.000000  21613.000000  21613.000000  \
mean      291.509045   1971.005136     84.402258  98077.939805     47.560053   
min         0.000000   1900.000000      0.000000  98001.000000     47.155900   
25%         0.000000   1951.000000      0.000000  98033.000000     47.471000   
50%         0.000000   1975.000000      0.000000  98065.000000     47.571800   
75%       560.000000   1997.000000      0.000000  98118.000000     47.678000   
max      4820.000000   2015.000000   2015.000000  98199.000000     47.777600   
std       442.575043     29.373411    401.679240     53.505026      0.138564   

               long  sqft_living15     sqft_lot15  
count  21613.000000   21613.000000   21613.000000  
mean    -122.213896    1986.552492   12768.455652  
min     -122.519000     399.000000     651.000000  
25%     -122.328000    1490.000000    5100.000000  
50%     -122.230000    1840.000000    7620.000000  
75%     -122.125000    2360.000000   10083.000000  
max     -121.315000    6210.000000  871200.000000  
std        0.140828     685.391304   27304.179631  

[8 rows x 21 columns]

#### можно сделать следующие выводы:
+ 21613 записей 
+ средняя цена 540 000 (с 75 000 до 7 700 000)
+ половина домов по цене не более 450 000
+ количество спальных с 0 до 33
+ половина проданных домов с 3 спальнями
+ количество ванных с 0 до 8
+ больше проданных домов с 2.5 ванных комнат
+ жилая площадь в размере от 290 m2 до 13 540 m2
+ половина домов жилой площадью не более 1 910 m2
+ общая площадь в размере от 520 m2 до 1 651 359 m2
+ половина домов общей площадью не более 7 618 m2
+ этажи с 1 по 3.5
+ половина домов этажностью до 1.5
+ в большинстве домов набережной нет (75% = 0)
+ хороший вид тоже у небольшого числа домов (75% = 0)
+ состояние домов измеряется от 1 до 5 (половина домов имеет коэф. = 3)
+ большинство домов со средним уровнем конструкции и дизайна
+ годы постройки с 1900 по 1997
+ есть информация о том, когда дом ремонтировался
+ указан почтовый индекс
+ географическое расположение (широта и долгота)
+ Кв. метры дома которые находятся выше земли от 290 до 2 210
+ Половина домов - дома с площадью части дома которая находится выше земли не более 9 410 m2
+ Кв. метры дома которые находятся ниже земли от 0 до 2 210
+ Половина домов - домас площадью части дома которая находится ниже земли не более 4 820 m2
+ Кв.метры жилой площади у 15 соседей
+ Кв.метры общей площади у 15 соседей

## Задание 3
### 3.1 В каком диапазоне изменяются стоимости недвижимости?
### 3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам?
### 3.3 Как много домов с разными этажами в данных?
### 3.4 Насколько хорошие состояния у домов в данных?
### 3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных?

In [8]:
print(f'Стоимость недвижимости меняется в дапозоне от {df.price.min()} до {df.price.max()}' )

Стоимость недвижимости меняется в дапозоне от 75000.0 до 7700000.0


In [9]:
print(f'В среднем жилая площадь занимает {round((df.sqft_living.sum()/df.sqft_lot.sum())*100,2)}% от всей площади')

В среднем жилая площадь занимает 13.77% от всей площади


In [10]:
df.floors.value_counts()

floors
1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
Name: count, dtype: int64

In [11]:
df.condition.value_counts()

condition
3    14031
4     5679
5     1701
2      172
1       30
Name: count, dtype: int64

In [12]:
df.condition.value_counts(normalize = True)

condition
3    0.649193
4    0.262759
5    0.078703
2    0.007958
1    0.001388
Name: proportion, dtype: float64

In [13]:
print(f'Первый дом построили в {df.yr_built.min()} году, последний в {df.yr_built.max()} году')

Первый дом построили в 1900 году, последний в 2015 году


## Задание 4
### 4.1 Сколько в среднем стоят дома, у которых 2 спальни?
### 4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000?
### 4.3 Как много домов коснулся ремонт?
### 4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?


In [14]:
print(f'Средняя цена дома с 2 спальнями = {round(df.price[df.bedrooms == 2].mean(),2)}')

Средняя цена дома с 2 спальнями = 401372.68


In [15]:
print(f'Средняя общая площадь домов, у которых стоимость более 600 000 = {round(df.sqft_lot[df.price > 600000].mean(),2)}m2')

Средняя общая площадь домов, у которых стоимость более 600 000 = 20442.52m2


In [16]:
print(f'Ремонт сделан в {df.yr_renovated[df.yr_renovated > 0].count()} домах')

Ремонт сделан в 914 домах


In [17]:
a = df.price[df.grade > 10].mean() - df.price[df.grade < 4].mean()
print(f'В среднем стоимость домов с оценкой grade выше 10 отличается от стоимости домов с оценкой grade меньше 4  на {round(a, 2) }')

В среднем стоимость домов с оценкой grade выше 10 отличается от стоимости домов с оценкой grade меньше 4  на 1488885.12


## Задание 5
### 5.1 Выберите дом клиенту
Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?
### 5.2 Выберите дом клиенту
Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?
### 5.3 Выберите дом клиенту
Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?


In [18]:
print(f'Вариантов в задании "5.1": {df[(df.waterfront > 0) & (df.bathrooms >= 3) & (df.sqft_basement > 0)].shape[0]}')

Вариантов в задании "5.1": 41


In [19]:
df['waterfront'].value_counts()

waterfront
0    21450
1      163
Name: count, dtype: int64

In [20]:
client = df[((df.view > 2) | (df.waterfront > 0)) & (df.condition > 3)  & (df.yr_built > 1979)]
print(f'Диапазон цен с условиями задания "5.2" : с {client.price.min()} по {client.price.max()}')

Диапазон цен с условиями задания "5.2" : с 420000.0 по 5300000.0


In [21]:
c = df.condition[(df.sqft_basement == 0) & (df.floors == 2) & (df.price < 150000) ].mean()
print(f'В среднем оценка по состоянию у домов с условиями задания "5.3" = {round(c, 2)}')

В среднем оценка по состоянию у домов с условиями задания "5.3" = 2.83
